### ISSUE TITLE
Generic hyperparameter scheduler 

### ISSUE THREAD

**Issue by arashb** [plumerai/larq#104]
The Idea is to have a TF scheduler which is optimizer/hyperparameter agnostic. 
As an example, it can be used for gamma, threshold or learning rate of the floating point optimizer in Bop.

**Comment by lgeiger** [comment]
Closed by #104
